lib/solargraph/source_map/mapper.rb

```diff
@@ -147,6 +147,23 @@ def process_directive source_position, comment_position, directive
               pins.last.docstring.add_tag YARD::Tags::Tag.new(:param, '', pins.last.return_type.to_s.split(', '), 'value')
             end
           end
+        when 'visibility'
+          begin
+            kind = directive.tag.text&.to_sym
+            return unless kind == :private || kind == :protected
```

Comment:
The possible values are 'public' | 'protected' | 'private'. In the case of public I don't see a case where the visibility would have to be adjusted. 

In the case of private it should be overwritten.
In the case of protected one could argue that if the visibility is private it should not overwrite the current pins?